crypto/conf/conf_def.h

```diff
@@ -28,21 +28,26 @@
 #define KEYTYPES(c)       ((const unsigned short *)((c)->meth_data))
 
 #ifndef CHARSET_EBCDIC
-# define CVT(a) ((a) & 0x7F)
+# define CVT(a) ((unsigned char)(a) <= 127 ? (unsigned char)(a) : 127)
 #else
-# define CVT(a) os_toascci[(a) & 0x7F]
+# define CVT(a) os_toascii[(unsigned char)(a)]
 #endif
 
-#define IS_COMMENT(c,a)     (KEYTYPES(c)[CVT(a)] & CONF_COMMENT)
-#define IS_FCOMMENT(c,a)    (KEYTYPES(c)[CVT(a)] & CONF_FCOMMENT)
-#define IS_EOF(c,a)         (KEYTYPES(c)[CVT(a)] & CONF_EOF)
-#define IS_ESC(c,a)         (KEYTYPES(c)[CVT(a)] & CONF_ESC)
-#define IS_NUMBER(c,a)      (KEYTYPES(c)[CVT(a)] & CONF_NUMBER)
-#define IS_WS(c,a)          (KEYTYPES(c)[CVT(a)] & CONF_WS)
-#define IS_ALNUM(c,a)       (KEYTYPES(c)[CVT(a)] & CONF_ALNUM)
-#define IS_ALNUM_PUNCT(c,a) (KEYTYPES(c)[CVT(a)] & CONF_ALNUM_PUNCT)
-#define IS_QUOTE(c,a)       (KEYTYPES(c)[CVT(a)] & CONF_QUOTE)
-#define IS_DQUOTE(c,a)      (KEYTYPES(c)[CVT(a)] & CONF_DQUOTE)
+/*
+ * Attention: because the macro argument 'a' is evaluated twice in CVT(a),
+ * it is not allowed pass 'a' arguments with side effects to IS_*(c,a)
+ * like for example IS_*(c, *p++).
```

Comment:
I'm sorry, but I think that introducing this kind of fragility is the wrong way to go.

Reply:
But note that this header file is used/included in only one internal source file.

Reply:
So we only shoot our own kneecaps off...  still.

Reply:
We're stuck with moving forward in pieces, I guess, because we had to remove the previous code.